### PR TITLE
12 api get articles by topic query

### PR DESCRIPTION
**Pull request by JenWarman**

### CODE REVIEW

File: models/topics.models.js [hunk 1]
```diff
@@ -29,23 +29,27 @@ exports.fetchArticleById = (article_id) => {
...
+        if (order === 'asc') {
+            queryString += ` ASC`
         }
         return db.query(queryString)
             .then(({ rows }) => {
                 return rows;
             })
     }
+    if (topic) {
+        return db.query(`SELECT * FROM articles WHERE topic=$1`, [topic])
```
Comment (nc-hannah-burks) [inline models/topics.models.js:45]: Rather than a separate query this should be added as a WHERE clause to the existing query (so a sortby and a order by could also be applied) 
 There should be a way of checking if the topic exists so we know the reason why if we get an empty array back (The existence of the topic should come from the db, not to be hard coded so it remains up to date)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -318,48 +318,89 @@ describe('nc-news API', () => {
...
+        test('200: responds with array of articles that match the topic query', () => {
+            return request(app)
+                .get('/api/articles?topic=cats')
+                .expect(200)
+                .then(({ body }) => {
+                    expect(body.articles).toHaveLength(1);
+                    body.articles.forEach((article) => {
+                        expect(article).toHaveProperty('topic', 'cats');
+                    })
+                })
```
Comment (nc-hannah-burks) [inline __tests__/app.test.js:387]: Another test to consider is Status 200, valid topic query, but has no articles responds with an empty array of articles, e.g. ?topic=paper.
